### ISSUE TITLE
Use Transmission RPC interface rather than SSH?

### ISSUE THREAD

**Issue by DaveCorder** [fm407/PIA-NextGen-PortForwarding#3]
Rather than setting up the SSH stuff to be able to ssh to the transmission server, it's much easier to just use the Transmission RPC API directly. Is there a reason you went with the ssh and transportupdate.sh script instead of direct RPC?

Here is some example code (which I added to the pfSense script and am currently using) to use the RPC API:

```
update_transmission_port () {
  # transmission user name
  TRANSUSER=transmission
  # transmission password
  TRANSPASS="RPCPasswordGoesHere"
  TRANSHOST=$TRANSIP

  SESSION_HEADER=$(curl -u $TRANSUSER:$TRANSPASS ${TRANSHOST}:9091/transmission/rpc 2>/dev/null | sed 's/.*<code>//g;s/<\/code>.*//g')

  if [ "x$SESSION_HEADER" != "x" ]; then
    DATA="{\"method\": \"session-set\", \"arguments\": { \"peer-port\" : $pf_port } }"
    echo $DATA
    CURLOUT=$(curl -u $TRANSUSER:$TRANSPASS --silent http://${TRANSHOST}:9091/transmission/rpc -d "$DATA" -H "$SESSION_HEADER")
    echo $CURLOUT
    echo "pia-nextgen-pf: $(date) Updated Transmission port number via RPC"
    logger "pia-nextgen-pf: Updated Transmission port number via RPC"
  else
    echo "pia-nextgen-pf: $(date) No Transmission RPC Session Header retrieved"
    logger "pia-nextgen-pf: No Transmission RPC Session Header retrieved"
  fi
}
```

**Comment by fm407** [comment]
oh beautiful! thank you for sharing, will test and implement, this is muuuuch better! thanks a lot for the contribution!

**Comment by Hakun1n** [comment]
Hey @DaveCorder good idea. Frankly I never thought about this.

@fm407 Recent draft pull request contains this as well. I've briefly tested it on my live system and it works :]

**Comment by fm407** [comment]
thanks for the PRs!

> Hey @DaveCorder good idea. Frankly I never thought about this.
> 
> @fm407 Recent draft pull request contains this as well. I've briefly tested it on my live system and it works :]

Thanks for the PRs! will let it run on my pfSense for today and then merge it!

**Comment by fm407** [comment]
implemented in #8
